### PR TITLE
LibreSSL: update to 2.4.3

### PR DESCRIPTION
**Pull request by jerome-benoit**
This update cover issues with weak ciphering and digesting algorithm that are now disabled by default. 

Signed-off-by: Jérôme Benoit jerome.benoit@piment-noir.org

### CODE REVIEW

**Comment by MilhouseVH** [comment]
I really don't want to have to keep bumping 5 files (package + 4 config files) each time we change the version, is there any way we can use a sed to rewrite the `Version: ` in each of the config files?

File: packages/security/libressl/package.mk [hunk 1]
```diff
@@ -17,23 +17,28 @@
...
 
 post_makeinstall_target() {
-  # backwards comatibility
   mkdir -p $INSTALL/etc/pki/tls
     ln -sf /etc/ssl/cert.pem $INSTALL/etc/pki/tls/cacert.pem
+  mkdir -p $INSTALL/etc/pki/tls/certs
+    ln -sf /etc/ssl/cert.pem $INSTALL/etc/pki/tls/certs/ca-bundle.crt
+
+  mkdir -p $SYSROOT_PREFIX/usr/lib/pkgconfig
+    cp $PKG_DIR/config/*.pc $SYSROOT_PREFIX/usr/lib/pkgconfig
```
Comment (MilhouseVH) [inline packages/security/libressl/package.mk]: If you change the hard-coded version in *.pc from `2.4.3` to `@PKG_VERSION@`, eg:
```
Version: @PKG_VERSION@
```
then use a `sed` to fix the *.pc files, ie. 
```
    sed -e "s/@PKG_VERSION@/$PKG_VERSION/g" -i $SYSROOT_PREFIX/usr/lib/pkgconfig/*.pc
```
and then we don't need to keep all the files in sync.

Not sure if applying the sed to all files in $SYSROOT_PREFIX/usr/lib/pkgconfig might produce unexpected results, in which case we should process the individual source files and output the new content to $SYSROOT_PREFIX/usr/lib/pkgconfig, ie:
```
     for f in $PKG_DIR/config/*.pc; do
       sed -e "s/@PKG_VERSION@/$PKG_VERSION/g" $f > $SYSROOT_PREFIX/usr/lib/pkgconfig/$(basename $f)
     done
```

**Comment by chewitt** [comment]
@jerome-benoit is there any change to the API/so version with this bump?

**Comment by stefansaraev** [comment]
you should not use cmake for building libressl, and you have any issues with those pkg-config files.

File: packages/security/libressl/package.mk (hunk 1 above)
Comment (jerome-benoit) [inline packages/security/libressl/package.mk]: I will replace the pkg-config libressl handling by a proper way of doing it in the next PR.

**Comment by jerome-benoit** [comment]
@stefansaraev : 

First it's not a switch to CMake just for the sake of switching. CMake is just way faster at building on SMP box in general and libressl in particular (two times faster here). 

Second, where to you see hacks on other packages related to this PR ?

**Comment by lrusak** [comment]
This won't work anyway cmake requires libressl to be built but libressl requires cmake.

Please leave it using autotools.

**Comment by stefansaraev** [comment]
the 4 pkg-config files added by this PR - that I call a hack

**Comment by stefansaraev** [comment]
you can of course disable PKG_AUTORECONF. libressl will compile fine, and a bit quicker than before. autoreconf is not needed for libressl at all

File: packages/security/libressl/package.mk [hunk 2]
```diff
@@ -17,23 +17,26 @@
...
-PKG_VERSION="2.3.8"
+PKG_VERSION="2.4.3"
 PKG_REV="1"
 PKG_ARCH="any"
 PKG_LICENSE="BSD"
 PKG_SITE="http://www.libressl.org/"
 PKG_URL="http://ftp.openbsd.org/pub/OpenBSD/LibreSSL/$PKG_NAME-$PKG_VERSION.tar.gz"
+PKG_DEPENDS_HOST="ccache:host"
 PKG_DEPENDS_TARGET="toolchain"
+PKG_PRIORITY="optional"
```
Comment (stefansaraev) [inline packages/security/libressl/package.mk]: PKG_PRIORITY is used nowhere. 

#719

File: packages/security/libressl/package.mk [hunk 3]
```diff
@@ -17,23 +17,26 @@
...
 PKG_IS_ADDON="no"
 PKG_USE_CMAKE="no"
-PKG_AUTORECONF="yes"
+PKG_AUTORECONF="no"
 
 post_makeinstall_target() {
-  # backwards comatibility
   mkdir -p $INSTALL/etc/pki/tls
     ln -sf /etc/ssl/cert.pem $INSTALL/etc/pki/tls/cacert.pem
+  mkdir -p $INSTALL/etc/pki/tls/certs
```
Comment (stefansaraev) [inline packages/security/libressl/package.mk]: this should be in a separate commit

**Comment by stefansaraev** [comment]
no soname change @chewitt it is safe.

**Comment by jerome-benoit** [comment]
None that I'm aware of.

-- 

Jérôme Benoit aka fraggle
Piment Noir - http://piment-noir.org
OpenPGP Key ID : 9FE9161D
Key fingerprint : 9CA4 0249 AF57 A35B 34B3 AC15 FAA0 CB50 9FE9 161D

File: packages/security/libressl/package.mk [hunk 4]
```diff
@@ -17,20 +17,21 @@
...
 
 PKG_NAME="libressl"
-PKG_VERSION="2.3.8"
+PKG_VERSION="2.4.3"
 PKG_REV="1"
 PKG_ARCH="any"
 PKG_LICENSE="BSD"
 PKG_SITE="http://www.libressl.org/"
 PKG_URL="http://ftp.openbsd.org/pub/OpenBSD/LibreSSL/$PKG_NAME-$PKG_VERSION.tar.gz"
+PKG_DEPENDS_HOST="ccache:host"
```
Comment (lrusak) [inline packages/security/libressl/package.mk]: what's the reason for this?

**Comment by jerome-benoit** [comment]
On Fri, 04 Nov 2016 14:56:43 -0700 in
<LibreELEC/LibreELEC.tv/pull/875/review/7291570@github.com>, Lukas
Rusak <notifications@github.com> wrote:


>  PKG_ARCH="any"
>  PKG_LICENSE="BSD"
>  PKG_SITE="http://www.libressl.org/"
>  PKG_URL="http://ftp.openbsd.org/pub/OpenBSD/LibreSSL/$PKG_NAME-$PKG_VERSION.tar.gz"
> +PKG_DEPENDS_HOST="ccache:host"
> 
> what's the reason for this?
> 

It's an explicit declaration of host ccache dependency instead of
an hidden implicit one. 

-- 
Jérôme Benoit aka fraggle
Piment Noir - http://piment-noir.org
OpenPGP Key ID : 9FE9161D
Key fingerprint : 9CA4 0249 AF57 A35B 34B3 AC15 FAA0 CB50 9FE9 161D

**Comment by lrusak** [comment]
thanks!

**Comment by jerome-benoit** [comment]
On Sat, 05 Nov 2016 09:52:56 -0700 in <LibreELEC/LibreELEC.tv/pull/875/c258624084@github.com>, 
Lukas Rusak <notifications@github.com> wrote:

> thanks!

You're welcome :)

-- 
Jérôme Benoit aka fraggle
Piment Noir - http://piment-noir.org
OpenPGP Key ID : 9FE9161D
Key fingerprint : 9CA4 0249 AF57 A35B 34B3 AC15 FAA0 CB50 9FE9 161D

**Comment by chewitt** [comment]
This commit increments the so version !!!!

I had to remove the tvheadend42 update that was build/pushed this morning as it will not run on currently released 7.90.001-008 builds. @jerome-benoit It is not a major issue to do this in alpha, but in future it would be nice to have proper warning (and I did ask) as users need to be warned of the change. There will need to be a full rebuild of add-ons with an SSL dependency when we release 7.90.009.

tvheadend42 built yesterday for testing:

```LE-CvH:~/.kodi/addons/service.tvheadend42/bin # ldd tvheadend
        linux-vdso.so.1 (0x00007ffe989ed000)
        libssl.so.38 => /usr/lib/libssl.so.38 (0x00007fc52e790000)
        libcrypto.so.37 => /usr/lib/libcrypto.so.37 (0x00007fc52e40f000)
        libz.so.1 => /usr/lib/libz.so.1 (0x00007fc52e77e000)
        libdl.so.2 => /lib/libdl.so.2 (0x00007fc52e20b000)
        libpthread.so.0 => /lib/libpthread.so.0 (0x00007fc52dfee000)
        librt.so.1 => /lib/librt.so.1 (0x00007fc52dde6000)
        libm.so.6 => /lib/libm.so.6 (0x00007fc52dae1000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007fc52d766000)
        libc.so.6 => /lib/libc.so.6 (0x00007fc52d3ca000)
        libresolv.so.2 => /lib/libresolv.so.2 (0x00007fc52d1b3000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fc52e5c4000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007fc52cf9d000)```

and the version (with this commit) build/pushed this morning:

```LE-CvH:~/.kodi/addons/service.tvheadend42/bin # ldd tvheadend
        linux-vdso.so.1 (0x00007ffea93a7000)
        libssl.so.39 => not found
        libcrypto.so.38 => not found
        libz.so.1 => /usr/lib/libz.so.1 (0x00007f8dab78f000)
        libdl.so.2 => /lib/libdl.so.2 (0x00007f8dab37d000)
        libpthread.so.0 => /lib/libpthread.so.0 (0x00007f8dab160000)
        librt.so.1 => /lib/librt.so.1 (0x00007f8daaf58000)
        libm.so.6 => /lib/libm.so.6 (0x00007f8daac53000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007f8daa8d8000)
        libc.so.6 => /lib/libc.so.6 (0x00007f8daa53c000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f8dab581000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007f8daa326000)```

**Comment by stefansaraev** [comment]
@chewitt sory for misleading you (I said there is no soname bump). I did indeed build test an earlier libressl version just to check if soname changes, but now I realize it was not 2.3.x what I tested, but an earlier 2.4.x ...

**Comment by chewitt** [comment]
I checked and 2.3.9 is the last version before the soname bump so I suggest we drop back to that for 8.0 builds. We can bump master again after we fork the 8.0 release branch.

**Comment by jerome-benoit** [comment]
On Sun, 06 Nov 2016 22:21:28 -0800 in
<LibreELEC/LibreELEC.tv/pull/875/c258756028@github.com>, Christian
Hewitt <notifications@github.com> wrote:

> I checked and 2.3.9 is the last version before the soname bump so I
> suggest we drop back to that for 8.0 builds. We can bump master again
> after we fork the 8.0 release branch.

Just revert the merge and either you reopen this PR if possible, either
I resubmit this PR for a future merge. 

-- 
Jérôme Benoit aka fraggle
Piment Noir - http://piment-noir.org
OpenPGP Key ID : 9FE9161D
Key fingerprint : 9CA4 0249 AF57 A35B 34B3 AC15 FAA0 CB50 9FE9 161D
